pkg/cloudprovider/aws/instance.go

```diff
@@ -104,31 +104,28 @@ func (p *InstanceProvider) Create(ctx context.Context,
 	return createFleetOutput.Instances[0].InstanceIds[0], nil
 }
 
-func (p *InstanceProvider) Terminate(ctx context.Context, nodes []*v1.Node) error {
-	if len(nodes) == 0 {
+func (p *InstanceProvider) Terminate(ctx context.Context, node *v1.Node) error {
+	if node == nil {
```

Comment:
I don't think you even need this nil check. Just make sure the code never calls Terminate(ctx,nil). This is defensive programming, which is great for libraries, but perhaps a bit overly verbose when we own all the code.

Reply:
You're right. This will only be triggered off a reconcile loop for a node, which means that we're guaranteed it's never nil. If it is, it will fail at the top level controller anyway. Good callout!